lib/views/pages/post/post_page_widget/bottom_bar_add_comment.dart

```diff
@@ -1,59 +1,105 @@
     required this.currentDisplayName,
   });
 
   @override
-  Widget build(BuildContext context) {
-    return Row(
-      // Align items to the start of the cross axis
-      crossAxisAlignment: CrossAxisAlignment.start,
-      children: [
-        Padding(
+  Widget build(BuildContext context, WidgetRef ref) {
+    final contentController = useTextEditingController();
+
+    final asyncNewCommentState = ref.watch(
+      newCommentStateProvider(parentPostId),
+    );
+
+    return CircularValue(
+      value: asyncNewCommentState,
+      builder: (context, newCommentState) {
+        // The avatar of the current user on the left
+        final userAvatar = Padding(
           padding: const EdgeInsets.only(right: 8),
           child: UserAvatar(
             key: commentUserAvatarKey,
             displayName: currentDisplayName,
             radius: 22,
           ),
-        ),
-        const Expanded(
+        );
+
+        // The field in which the user can write a comment
+        final commentTextField = Expanded(
           child: TextField(
             key: addCommentTextFieldKey,
             minLines: 1,
             maxLines: 5,
             decoration: InputDecoration(
-              contentPadding: EdgeInsets.all(8),
-              border: OutlineInputBorder(),
+              contentPadding: const EdgeInsets.all(8),
+              border: const OutlineInputBorder(),
               hintText: _textFieldHintAddComment,
+              errorText: newCommentState.contentError,
             ),
+            controller: contentController,
           ),
-        ),
-        Align(
+        );
+
+        // The button to post the comment
+        final addCommentButton = Align(
           // Keeps the IconButton centered in the cross axis
           alignment: Alignment.center,
           child: IconButton(
             key: postCommentButtonKey,
             icon: const Icon(Icons.send),
-            onPressed: () {
-              //TODO: handle add comment
-              FocusManager.instance.primaryFocus?.unfocus();
+            onPressed: () async {
+              final newCommentViewModel = ref.read(
+                newCommentStateProvider(parentPostId).notifier,
+              );
+
+              final commentPosted = await newCommentViewModel.tryAddComment(
+                contentController.text,
+              );
+
+              // If the comment was posted, refresh the comment list
+              // and reset the new comment view model
+              if (commentPosted) {
+                await ref
+                    .read(commentListProvider(parentPostId).notifier)
+                    .refresh();
+                await newCommentViewModel.reset();
+                contentController.clear();
+                FocusManager.instance.primaryFocus?.unfocus();
+              }
             },
           ),
-        ),
-      ],
+        );
+
+        return Row(
+          // Align items to the start of the cross axis
```

Comment:
I think we should avoid comments that paraphrase the code. We can read the same information on the next line. (more on the picky side I admit)